src/Data/Array/Mutable/Linear.hs

```diff
@@ -215,6 +220,20 @@ slice from targetSize arr =
     wrap :: (# Array# a, Array# a  #) #-> (Array a, Array a)
     wrap (# old, new #) = (Array old, Array new)
 
+-- | /O(1)/ Convert an 'Array' to an immutable 'Vector.Vector' (from
+-- 'vector' package).
+freeze :: Array a #-> Ur (Vector.Vector a)
+freeze (Array arr) =
+  Unlifted.freeze go arr
+ where
+   go arr = unsafeDupablePerformIO $ do
+     mut <- Prim.unsafeThawArray (Prim.Array arr)
+     let mv = MVector.MVector 0 (Prim.sizeofMutableArray mut) mut
```

Comment:
I think this is correct because of the source of the [MVector instance for MVector](https://hackage.haskell.org/package/vector-0.12.1.2/docs/src/Data.Vector.Mutable.html#line-86).